src/lib/generateDefinition.ts

```diff
@@ -0,0 +1,75 @@
+import {Cluster} from 'zigbee-herdsman/dist/zcl/tstype';
+import {Definition, ModernExtend, Zh} from './types';
+import * as e from './modernExtend';
+import {Endpoint} from 'zigbee-herdsman/dist/controller/model';
+
+export function generateDefinition(device: Zh.Device): Definition {
```

Comment:
This generator should also return the code which can be used to fully support this device. e.g.

```
{
  model: 'bla',
  vendor: 'bla',
  description: 'the description',
  extend: [onOff()],
}
```

Reply:
I am not that well versed in JS for this, but I suppose `extend` options can be strigified (somehow) and added to definition as `extend_source` or similar field to result in `extend_source: ['e.pressure()', 'e.onOff({powerOnBehavior: false})']`. When needed - single quotes can be removed to result in callable code.

Other information(model, vendor, description) will be available from device definition, or otherwise user updated/generated.

Reply:
I would return the whole source here, also the imports needs to be returned (because for e.g. Philips light we are going to use `philipsLight()` and not `light()`

Reply:
Do you think that this may be left for next iteration? I think this might take a while to do properly for me.

Especially if you mean to return _verbatim_
```
{
  model: 'bla',
  vendor: 'bla',
  description: 'the description',
  // I don't know how to return this without executing `onOff`,
  // which could still expand to function call(i.e. for `configure`).
  extend: [onOff()],
}
```

For this only thing that I can think of for now is to return something like
```
type extender = {
  functionName: string, // i.e. `onOff`
  args: Object, // i.e. `{powerOnBehavior: true}`
  importedFrom: string, // empty for default, or path relative to `generateDefinition.ts`. I.e. `./philips`.
}
```

What do you have in mind?

Reply:
Lets leave it out for now, I propose to merge this now with the functionality being disabled for now. I already started working on integrating the generation of external definitions through the frontend.

Reply:
I've updated all I could, so I will stop updating this PR until and if you would need me to.